src/readcalls.c

```diff
@@ -197,206 +195,30 @@ int readcalls(const char *logfile) {
 	show_progress(linenr);
 
 	if (log_is_comment(inputbuffer))
-	    continue;		/* skip note in  log  */
-
-	// prepare helper variables
-	pfxnumcntidx = -1;
-	pxnr = 0;
+	    continue;		/* skip note in log */
 
 	qso = parse_qso(inputbuffer);
-	bandindex = qso->bandindex;
 
 	/* get the country number, not known at this point */
 	countrynr = getctydata(qso->call);
 
-	/*  lookup worked stations, add if new */
-	int station = lookup_or_add_worked(qso->call);
-	update_worked(station, qso);
-
-
-	if (continentlist_only) {
-	    if (!is_in_continentlist(dxcc_by_index(countrynr)->continent)) {
-		qsos_per_band[bandindex]++;
-		free_qso(qso);
-		qso = NULL;
-		continue;
-	    }
-	}
-
-	if (iscontest) {
-	    // get points
-	    total = total + log_get_points(inputbuffer);
-
-	    if (CONTEST_IS(CQWW) || itumult || wazmult) {
-		// get the zone
-		z = zone_nr(inputbuffer + 54);
-	    }
-
-	    if (wysiwyg_once || wysiwyg_multi ||
-		    unique_call_multi != 0 ||
-		    CONTEST_IS(ARRL_SS) ||
-		    serial_section_mult ||
-		    serial_grid4_mult ||
-		    sectn_mult ||
-		    sectn_mult_once ||
-		    (dx_arrlsections
-		     && ((countrynr == w_cty) || (countrynr == ve_cty)))) {
-		// get multi info
-		char *multbuffer = get_multi_from_line(inputbuffer);
-		remember_multi(multbuffer, bandindex, 0);
-		g_free(multbuffer);
-	    }
-	}
-
-
-	if (pfxmultab) {
-	    getpx(qso->call);
-	    add_pfx(wpx_prefix, bandindex);
-	}
-
-
-	/* look if calls are excluded */
-	add_ok = true;
-
-	if (CONTEST_IS(ARRLDX_USA)
-		&& ((countrynr == w_cty) || (countrynr == ve_cty)))
-	    add_ok = false;
-
-	if (CONTEST_IS(PACC_PA)) {
-
-	    strcpy(hiscall, qso->call);
-
-	    add_ok = pacc_pa();
-
-	    if (add_ok == false) {
-		qsos_per_band[bandindex]++;
-	    }
-
-	    hiscall[0] = '\0';
-	}
-
-	if (pfxnummultinr > 0) {
-	    getpx(qso->call);
-	    pxnr = districtnumber(wpx_prefix);
-
-	    getctydata(qso->call);
-
-	    pfxnumcntidx = lookup_country_in_pfxnummult_array(countrynr);
-
-	    add_ok = true;
-	}
-
-	veto = check_veto(countrynr);
-
-	if (add_ok) {
+	checkexchange(qso->comment, false);
+	dupe = is_dupe(qso->call, qso->bandindex, qso->mode);
+	addcall(qso);
+	score_qso();
 
-	    worked[station].band |= inxes[bandindex];	/* mark band as worked */
-
-	    qsos_per_band[bandindex]++;
-
-	    if (CONTEST_IS(CQWW) || itumult || wazmult)
-		zones[z] |= inxes[bandindex];
-
-	    if (pfxnumcntidx < 0) {
-		if (!veto) {
-		    countries[countrynr] |= inxes[bandindex];
-		}
-	    } else {
-		pfxnummulti[pfxnumcntidx].qsos[pxnr] |= inxes[bandindex];
-	    }
-	}
 	free_qso(qso);
 	qso = NULL;
     }
 
     fclose(fp);
 
-    /* all lines red, now build other statistics */
```

Comment:
All this must also happen during `addcall()` & co. Will add some tests to check it.

Reply:
addcall() should be fine in building the statistics. The statistics code at the end of readcalls() were written so as to mimic the logic represented by addcall().